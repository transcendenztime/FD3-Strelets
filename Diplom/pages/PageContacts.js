"use strict";

import React from 'react';
import { NavLink } from 'react-router-dom';

import { checkNameValue, checkEmailValue, checkTelValue, checkTextValue  } from '../services/checkForm';

import '../styles/pages/Contacts.css';

class PageContacts extends React.PureComponent {
          
  state = {
    name: '',
    email: '',
		tel: '',
		text: '',
    errorName: 0, //0-нет ошибки, 1-пустое значение, 2-неверное значение
    errorEmail: 0, //0-нет ошибки, 1-нет ошибки тк не обязат поле, 2-неверное значение
		errorTel: 0, //0-нет ошибки, 1-пустое значение, 2-неверное значение
		errorText:0, //0-нет ошибки, 1-пустое значение

  }
  
  changeName = (e)=> {
    this.setState( {name: e.target.value, errorName: 0} );
  };
  checkName = () => {
    let check = checkNameValue(this.state.name);
    this.setState( {errorName: check} );
  };

  changeEmail = (e)=> {
    this.setState( {email: e.target.value, errorEmail: 0} );
  };
  checkEmail = () => {
    let check = checkEmailValue(this.state.email);
    this.setState( {errorEmail: check} );
  };

  changeTel = (e)=> {
    this.setState( {tel: e.target.value, errorTel: 0} );
  };
  checkTel = () => {
    let check = checkTelValue(this.state.tel);
    this.setState( {errorTel: check} );
	};
	
  changeText = (e)=> {
    this.setState( {text: e.target.value, errorText: 0} );
  };
  checkText = () => {
    let check = checkTextValue(this.state.text);
    this.setState( {errorText: check} );
  };

  checkForm = (e) => {
    this.checkName();
    this.checkTel();
    this.checkText();

    if(this.state.errorName || this.state.errorEmail || this.state.errorTel || this.state.errorText) {
      e.preventDefault();
    }
	}
	

  render() {

    return (
      <div className = "contacts page">

        <h1 className="page-title">Контакты:</h1>

        <div className = "breadcrumbs-container">
          <NavLink to="/" exact className="breadcrumbs">Главная </NavLink>
          <span className="breadcrumbs-arr" > &rarr; </span>
          <NavLink to="/contacts" className="breadcrumbs">Контакты</NavLink>
        </div>
        <hr />

        <div className="content__contacts">
					<h3 className="page-subtitle"> Время работы магазина:</h3>
					<p className="text">понедельник - пятница: с 9.00 до 18.00 </p>
					<h3 className="page-subtitle"> Адрес магазина:</h3>
					<p className="text">г. Минск, ул. Змитрока Бядули, 15</p>
					<h3 className="page-subtitle"> Телефон:</h3>
					<p className="text">+375 (33) 325-50-11</p>
          <p className = "text">Прием заказов по телефону с 9.00 до 18.00</p>
					<p className="text">Заказы через сайт принимаются круглосуточно.</p>
				</div>    
				    
				<h2 className="page-subtitle"> Карта проезда:</h2>
        <iframe src="https://yandex.ru/map-widget/v1/?um=constructor%3A8d49deaeefa4b69aeec612972758a0dafd091fb16774e7a48873fbabbf0cca52&amp;source=constructor" width="600" height="450" frameBorder="0" allowFullScreen className = "contacts-map"></iframe>
        
				<div className="contacts__feedback">
					<h2 className="contacts-subtitle">Обратная связь</h2>
					<hr />
					<form className="contacts__feedback-form"  onSubmit = {this.checkForm}>
						<div className = "feedback-form__container">

							<div>
								<label htmlFor="feedback-name" className="contacts-title">Имя *</label><br />
								<input type="text" name="feedback-name" id="feedback-name" className="feedback-input"
									onChange = {this.changeName} onBlur = {this.checkName} /><br />
								<span className = {this.state.errorName == 1 ? "visible" : "invisible"}> * Обязательное поле</span>
								<span className = {this.state.errorName == 2 ? "visible" : "invisible"}> * Введите до 30 буквенных символов</span>
							</div>

							<div>
								<label htmlFor="feedback-email" className="contacts-title">email</label><br />
								<input type="email" name="feedback-email" id="feedback-email" className="feedback-input"
									onChange = {this.changeEmail} onBlur = {this.checkEmail}  /><br />
								<span className = {this.state.errorEmail == 2 ? "visible" : "invisible"}> * Некорректный email </span>
							</div>
							
							<div>
								<label htmlFor="feedback-tel" className="contacts-title">Телефон *</label><br />
								<input type="tel" name="feedback-tel" id="feedback-tel" className="feedback-input"
									onChange = {this.changeTel} onBlur = {this.checkTel}  /><br />
								<span className = {this.state.errorTel == 1 ? "visible" : "invisible"}> * Обязательное поле</span>
								<span className = {this.state.errorTel == 2 ? "visible" : "invisible"}> * Некорректный номер телефона</span>
							</div>

							<div>
								<label htmlFor="feedback-text" className="contacts-title">Комментарий *</label><br />
								<textarea name="feedback-text" id="feedback-text" className = "feedback-textarea"
									onChange = {this.changeText} onBlur = {this.checkText}  ></textarea><br />										
								<span className = {this.state.errorText == 1 ? "visible" : "invisible"}> * Обязательное поле</span>
							</div>
						</div> <br />

						<div className = "feedback-submit-container">
							<input type="submit" name="feedback-submit" value="Отправить" className="feedback-submit" onClick = {this.checkForm}></input>
						</div>

					</form>
				</div>
        
      </div>
    );
    
  }

}
    
export default PageContacts;
    