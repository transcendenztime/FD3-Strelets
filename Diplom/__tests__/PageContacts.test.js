import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import PageContacts from '../pages/PageContacts';

test('внешний вид PageContacts', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
      <Router>
          <PageContacts />
      </Router>
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
          
  });