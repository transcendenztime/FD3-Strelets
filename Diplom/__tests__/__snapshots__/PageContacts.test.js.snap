// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`внешний вид PageContacts 1`] = `
<div
  className="contacts page"
>
  <h1
    className="page-title"
  >
    Контакты:
  </h1>
  <div
    className="breadcrumbs-container"
  >
    <a
      aria-current={null}
      className="breadcrumbs"
      href="/"
      onClick={[Function]}
      style={undefined}
    >
      Главная 
    </a>
    <span
      className="breadcrumbs-arr"
    >
       → 
    </span>
    <a
      aria-current={null}
      className="breadcrumbs"
      href="/contacts"
      onClick={[Function]}
      style={undefined}
    >
      Контакты
    </a>
  </div>
  <hr />
  <div
    className="content__contacts"
  >
    <h3
      className="page-subtitle"
    >
       Время работы магазина:
    </h3>
    <p
      className="text"
    >
      понедельник - пятница: с 9.00 до 18.00 
    </p>
    <h3
      className="page-subtitle"
    >
       Адрес магазина:
    </h3>
    <p
      className="text"
    >
      г. Минск, ул. Змитрока Бядули, 15
    </p>
    <h3
      className="page-subtitle"
    >
       Телефон:
    </h3>
    <p
      className="text"
    >
      +375 (33) 325-50-11
    </p>
    <p
      className="text"
    >
      Прием заказов по телефону с 9.00 до 18.00
    </p>
    <p
      className="text"
    >
      Заказы через сайт принимаются круглосуточно.
    </p>
  </div>
  <h2
    className="page-subtitle"
  >
     Карта проезда:
  </h2>
  <iframe
    allowFullScreen={true}
    className="contacts-map"
    frameBorder="0"
    height="450"
    src="https://yandex.ru/map-widget/v1/?um=constructor%3A8d49deaeefa4b69aeec612972758a0dafd091fb16774e7a48873fbabbf0cca52&source=constructor"
    width="600"
  />
  <div
    className="contacts__feedback"
  >
    <h2
      className="contacts-subtitle"
    >
      Обратная связь
    </h2>
    <hr />
    <form
      className="contacts__feedback-form"
      onSubmit={[Function]}
    >
      <div
        className="feedback-form__container"
      >
        <div>
          <label
            className="contacts-title"
            htmlFor="feedback-name"
          >
            Имя *
          </label>
          <br />
          <input
            className="feedback-input"
            id="feedback-name"
            name="feedback-name"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
          />
          <br />
          <span
            className="invisible"
          >
             * Обязательное поле
          </span>
          <span
            className="invisible"
          >
             * Введите до 30 буквенных символов
          </span>
        </div>
        <div>
          <label
            className="contacts-title"
            htmlFor="feedback-email"
          >
            email
          </label>
          <br />
          <input
            className="feedback-input"
            id="feedback-email"
            name="feedback-email"
            onBlur={[Function]}
            onChange={[Function]}
            type="email"
          />
          <br />
          <span
            className="invisible"
          >
             * Некорректный email 
          </span>
        </div>
        <div>
          <label
            className="contacts-title"
            htmlFor="feedback-tel"
          >
            Телефон *
          </label>
          <br />
          <input
            className="feedback-input"
            id="feedback-tel"
            name="feedback-tel"
            onBlur={[Function]}
            onChange={[Function]}
            type="tel"
          />
          <br />
          <span
            className="invisible"
          >
             * Обязательное поле
          </span>
          <span
            className="invisible"
          >
             * Некорректный номер телефона
          </span>
        </div>
        <div>
          <label
            className="contacts-title"
            htmlFor="feedback-text"
          >
            Комментарий *
          </label>
          <br />
          <textarea
            className="feedback-textarea"
            id="feedback-text"
            name="feedback-text"
            onBlur={[Function]}
            onChange={[Function]}
          />
          <br />
          <span
            className="invisible"
          >
             * Обязательное поле
          </span>
        </div>
      </div>
       
      <br />
      <div
        className="feedback-submit-container"
      >
        <input
          className="feedback-submit"
          name="feedback-submit"
          onClick={[Function]}
          type="submit"
          value="Отправить"
        />
      </div>
    </form>
  </div>
</div>
`;
